client/Dockerfile

```diff
@@ -0,0 +1,48 @@
+FROM debian:11
+
+RUN apt-get update && \
+    apt-get install -y git gnupg curl && \
+    apt-get clean
+
+ENV USER_ID=1001
+ENV HOME=/root
+
+WORKDIR ${HOME}
+
+COPY ./ ${HOME}/
+
+RUN git clone https://github.com/asdf-vm/asdf.git ${HOME}/asdf --depth 1 --branch v0.14.0
+
+ENV BASH_ENV="${HOME}/asdf/asdf.sh"
+# Because asdf is loaded via BASH_ENV, all commands using adsf need to be executed using /usr/bin/env bash -c
+SHELL ["/usr/bin/env", "bash", "-c"]
+
+COPY .tool-versions ${HOME}/.tool-versions
+# The client only needs yarn and node
+RUN sed -i -nr '/node|yarn/p' ${HOME}/.tool-versions && \
+    cat ${HOME}/.tool-versions | cut -f 1 -d ' ' | xargs -n 1 asdf plugin-add && \
+    asdf plugin-update --all && \
+    asdf install && \
+    asdf reshim
+
+ADD https://github.com/Yelp/dumb-init/releases/download/v1.2.5/dumb-init_1.2.5_x86_64 /usr/local/bin/dumb-init
+RUN chmod +x /usr/local/bin/dumb-init
+ENTRYPOINT ["dumb-init", "--"]
+
+ENV NODE_ENV=production
+
+RUN yarn install --frozen-lockfile --production=false && \
+    yarn build && \
+    yarn install --frozen-lockfile --production=true && \
```

Comment:
What is the reasoning behind running yarn install with production set to false, and then to true?

Reply:
I was kind of wondering the same, it's setup the same way in the registration app.
I'm assuming it's to ensure the development build works as well? Maybe @dleard or @andrea-williams has context to give us

Reply:
Unless their docs are off, the only difference is that with `production=false` our dev dependencies will be installed. In that case the second install won't do anything. If we need something from our dev dependencies for the build to happen well, we could consider removing the `node_modules` after the `yarn build` and then running the install in production mode.

Definitely not a huge deal, I was just curious.

Reply:
I'd say - let's remove the whole dev statement. There is no need to ship a production image with dev dependencies, this might increase the attack surface if those deps have vulnerabilities?

Reply:
I agree, and if nothing else we'll shrink our image a little bit, which feels like a win win to me!

Reply:
I'm afraid I have no insight into why the Dockerfile is written that way. It's a question for either Dylan or @Sepehr-Sobhani

Reply:
This seems to be a leftover from copying and pasting the Dockerfile from CIF.